packages/dds/tree/src/simple-tree/arrayNodeTypes.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeArrayNode } from "./arrayNode.js";
+import type {
+	ImplicitAllowedTypes,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+} from "./schemaTypes.js";
+import {
+	NodeKind,
+	type TreeNodeSchemaClass,
+	type TreeNodeSchema,
+	type TreeNodeSchemaNonClass,
+	type WithType,
+} from "./core/index.js";
+
+import type { SimpleArrayNodeSchema } from "./simpleSchema.js";
+
+/**
+ * A schema for customizable {@link (TreeArrayNode:interface)}s.
+ * @sealed
+ * @alpha
+ */
+export interface ArrayNodeCustomizableSchema<
```

Comment:
Should these interfaces also be `@system`? Do we want users to interact with these directly?

Reply:
We support narrowing to ArrayNodeCustomizableSchema | ArrayNodePojoEmulationSchema and I expect people walking schema to use that, so I think these are non-system.

They also provide a better way to refer to thinks like the insertable typed for an ArrayNode that the existing approach (using ReturnType on SchemaFactory.Array I guess?)

Reply:
I have added a ArrayNodeSchema type (thats non system) and marked these as system as their main use it output from our factory APis.